### PR TITLE
Made Reducer Function Comment More Explicit

### PR DESCRIPTION
**Pull request by bkillenit**
Increased comprehensibility of the comments for the reducer function to make the optimizations around the arguments.length value more understandable to contribs.

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -185,8 +185,8 @@
 
   // Create a reducing function iterating left or right.
   var createReduce = function(dir) {
-    // Optimized iterator function as using arguments.length
-    // in the main function will deoptimize the, see #1991.
+    // Optimize iterator function to use arguments.length
```
Comment (megawac) [inline underscore.js]: to avoid using arguments.length

File: underscore.js [hunk 2]
```diff
@@ -185,8 +185,8 @@
...
+    // in the main function. This will deoptimize the length lookup time.
```
Comment (megawac) [inline underscore.js]: , as this would deoptimize reduce, see #1991.

**Comment by megawac** [comment]
You're currently mixing up functions, that functions purpose is to avoid looking up arguments.length in the main reduce funciton

**Comment by jashkenas** [comment]
Yeah, this is backwards.

**Pull request closed without merging**